src/tests/unit/sources.rs

```diff
@@ -61,9 +57,9 @@ pub async fn add_source_with_invalid_name() {
 }
 
 #[actix_rt::test]
-pub async fn add_source_with_invalid_url() {
+pub async fn when_add_source_with_invalid_url_expect_error() {

```

Comment:
Should be be expecting a ResourecError::InvalidInput error here? `when_add_source_with_invalid_url_expect_invalid_input`?